axa-file-upload: 1 end-to-end test fails

The end-to-end test in ui.test.js fails in line 188 under Travis (see screenshot).
An attempt to add 50 ms waiting time after line 180 did not fix the problem.

Please investigate and fix this.

![Screen Shot 2019-09-16 at 15 21 17](https://user-images.githubusercontent.com/46926507/64961451-d5311080-d895-11e9-97f3-c96b7b545de8.png)

I just had another failed UI Test @NoahLiechti .
https://travis-ci.org/axa-ch/patterns-library/jobs/606033468?utm_medium=notification&utm_source=github_status

The other test was running??
![image](https://user-images.githubusercontent.com/7448004/68034546-21a7a280-fcc2-11e9-8eed-e8bd2b7c6e1a.png)


```

File upload
 ✓ should render file-upload
 ✖ should upload file and switch view
   1) AssertionError: expected false to be truthy
      Browser: HeadlessChrome 62.0.3202 / Linux 0.0.0
         39 |  const $fileOverviewElem = await Selector(() =>
         40 |    document
         41 |      .querySelector('axa-file-upload')
         42 |
      .shadowRoot.querySelector('.js-file-upload__dropzone-file-overview')
         43 |  );
       > 44 |  await t.expect($fileOverviewElem.exists).ok();
         45 |});
         46 |
         47 |test("shouldn't upload file because it's an invalid type",
      async t => {
         48 |  const $inputFileInputElem = await Selector(() =>
         49 |    document
         at ok
      (/home/travis/build/axa-ch/patterns-library/src/components/20-molecules/file-upload/ui.test.js:44:44)
```

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -177,7 +177,8 @@ test('should exceed maximum number of files', async t => {
 
   await t.expect($inputFileInputElem.exists).ok();
 
-  await t.setFilesToUpload($inputFileInputElem, validFiles);
+  await t.setFilesToUpload($inputFileInputElem, validFiles[0]);
+  await t.setFilesToUpload($inputFileInputElem, validFiles[1]);
 
   const $figureElems = await Selector(
     () => document.querySelector('axa-file-upload').shadowRoot

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -185,7 +185,7 @@ test('should exceed maximum number of files', async t => {
     () => document.querySelector('axa-file-upload').shadowRoot
   ).find('.js-file-upload__img-figure');
 
-  t.wait(1);
+  await $figureElems();
   await t.expect(await $figureElems.count).eql(1);
 
   const $errorWrapper = await Selector(() =>

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -149,7 +149,8 @@ test('should exceed maximum size of single file', async t => {
   await t.expect($fileName.innerText).eql('File size exceeds maximum size');
 });
 
-fixture('File upload - maxNumberOfFiles prop').page(
+// eslint-disable-next-line jest/no-focused-tests
+fixture.only('File upload - maxNumberOfFiles prop').page(
   `${host}/iframe.html?id=molecules-file-upload--file-upload&knob-inputFileText=Upload%20file&knob-maxSizeOfSingleFileKB=100&knob-maxSizeOfAllFilesKB=500&knob-maxNumberOfFiles=1&knob-deleteStatusText=Delete&knob-addStatusText=Add%20more&knob-fileTooBigStatusText=File%20size%20exceeds%20maximum%20size&knob-filesTooBigStatusText=File%20sizes%20exceed%20maximum%20size&knob-tooManyFilesStatusText=You%20exceeded%20the%20maximum%20number%20of%20files&knob-orText=or&knob-infoText=Drag%20and%20drop%20to%20upload%20your%20file&knob-icon=cloud-upload&knob-headerText=The%20following%20files%20are%20being%20transferred:`
 );
 test('should remove addMoreInputFile', async t => {
@@ -170,8 +171,7 @@ test('should remove addMoreInputFile', async t => {
   await t.expect($addMoreInputFieldElem.exists).notOk();
 });
 
-// eslint-disable-next-line jest/no-focused-tests
-test.only('should exceed maximum number of files', async t => {
+test('should exceed maximum number of files', async t => {
   const $inputFileInputElem = await Selector(
     () => document.querySelector(FILE_UPLOAD_TAG).shadowRoot,
     { dependencies: { FILE_UPLOAD_TAG } }

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -84,42 +84,40 @@ test('should convert .png file to .jpg', async t => {
     .ok();
 });
 
-// TODO: Travis fails at this test; Probably due performance issues (for local testing uncomment this section)
+test('should delete image', async t => {
+  const $inputFileInputElem = await Selector(() =>
+    document
+      .querySelector('axa-file-upload')
+      .shadowRoot.querySelector('axa-input-file .a-input-file__input')
+  );
+  await t.expect($inputFileInputElem.exists).ok();
 
-// test('should delete image', async t => {
-//   const $inputFileInputElem = await Selector(() =>
-//     document
-//       .querySelector('axa-file-upload')
-//       .shadowRoot.querySelector('axa-input-file .a-input-file__input')
-//   );
-//   await t.expect($inputFileInputElem.exists).ok();
+  await t.setFilesToUpload($inputFileInputElem, validFiles[1]);
 
-//   await t.setFilesToUpload($inputFileInputElem, validFiles[1]);
+  const $figureElemsBefore = await Selector(() =>
+    document
+      .querySelector('axa-file-upload')
+      .shadowRoot.querySelectorAll('.js-file-upload__img-figure')
+  );
 
-//   const $figureElemsBefore = await Selector(() =>
-//     document
-//       .querySelector('axa-file-upload')
-//       .shadowRoot.querySelectorAll('.js-file-upload__img-figure')
-//   );
+  await t.expect(await $figureElemsBefore.count).eql(2); // one file + addMoreInputFile
 
-//   await t.expect($figureElemsBefore.count).eql(2); // one file + addMoreInputFile
+  const $figureElem = await Selector(() =>
+    document
+      .querySelector('axa-file-upload')
+      .shadowRoot.querySelector('.js-file-upload__img-figure')
+  );
 
-//   const $figureElem = await Selector(() =>
-//     document
-//       .querySelector('axa-file-upload')
-//       .shadowRoot.querySelector('.js-file-upload__img-figure')
-//   );
+  await t.click($figureElem);
 
-//   await t.click($figureElem);
+  const $figureElemsAfter = await Selector(() =>
+    document
+      .querySelector('axa-file-upload')
+      .shadowRoot.querySelectorAll('.js-file-upload__img-figure')
+  );
 
-//   const $figureElemsAfter = await Selector(() =>
-//     document
-//       .querySelector('axa-file-upload')
-//       .shadowRoot.querySelectorAll('.js-file-upload__img-figure')
-//   );
-
-//   await t.expect($figureElemsAfter.count).eql(0); // zero because start view has no figures
-// });
+  await t.expect(await $figureElemsAfter.count).eql(0); // zero because start view has no figures
+});
 
 fixture('File upload - maxSizeOfSingleFileKB prop').page(
   `${host}/iframe.html?id=molecules-file-upload--file-upload&knob-inputFileText=Upload%20file&knob-maxSizeOfSingleFileKB=1&knob-maxSizeOfAllFilesKB=500&knob-maxNumberOfFiles=10&knob-deleteStatusText=Delete&knob-addStatusText=Add%20more&knob-fileTooBigStatusText=File%20size%20exceeds%20maximum%20size&knob-filesTooBigStatusText=File%20sizes%20exceed%20maximum%20size&knob-tooManyFilesStatusText=You%20exceeded%20the%20maximum%20number%20of%20files&knob-orText=or&knob-infoText=Drag%20and%20drop%20to%20upload%20your%20file&knob-icon=cloud-upload&knob-headerText=The%20following%20files%20are%20being%20transferred:`
@@ -141,7 +139,7 @@ test('should exceed maximum size of single file', async t => {
       .shadowRoot.querySelectorAll('.js-file-upload__img-figure')
   );
 
-  await t.expect($figureElems.count).eql(2); // one file + addMoreInputFile
+  await t.expect(await $figureElems.count).eql(2); // one file + addMoreInputFile
   const $fileName = await Selector(() =>
     document
       .querySelector('axa-file-upload')
@@ -171,34 +169,32 @@ test('should remove addMoreInputFile', async t => {
   await t.expect($addMoreInputFieldElem.exists).notOk();
 });
 
-// TODO: Travis fails at this point; Probably due performance issues (for local testing uncomment this section)
-// test('should exceed maximum number of files', async t => {
-//   const $inputFileInputElem = await Selector(() =>
-//     document
-//       .querySelector('axa-file-upload')
-//       .shadowRoot.querySelector('axa-input-file .a-input-file__input')
-//   );
-//   await t.expect($inputFileInputElem.exists).ok();
+test('should exceed maximum number of files', async t => {
+  const $inputFileInputElem = await Selector(
+    () => document.querySelector(FILE_UPLOAD_TAG).shadowRoot,
+    { dependencies: { FILE_UPLOAD_TAG } }
+  ).find('axa-input-file .a-input-file__input');
 
-//   await t.setFilesToUpload($inputFileInputElem, validFiles);
+  await t.expect($inputFileInputElem.exists).ok();
 
-//   const $figureElems = await Selector(() =>
-//     document
-//       .querySelector('axa-file-upload')
-//       .shadowRoot.querySelectorAll('.js-file-upload__img-figure')
-//   );
-//   await t.expect($figureElems.count).eql(1);
+  await t.setFilesToUpload($inputFileInputElem, validFiles);
 
-//   const $errorWrapper = await Selector(() =>
-//     document
-//       .querySelector('axa-file-upload')
-//       .shadowRoot.querySelectorAll('.js-file-upload__error-wrapper')
-//   );
+  const $figureElems = await Selector(
+    () => document.querySelector('axa-file-upload').shadowRoot
+  ).find('.js-file-upload__img-figure');
 
-//   await t
-//     .expect($errorWrapper.innerText)
-//     .eql('You exceeded the maximum number of files');
-// });
+  await t.expect(await $figureElems.count).eql(1);
+
+  const $errorWrapper = await Selector(() =>
+    document
+      .querySelector('axa-file-upload')
+      .shadowRoot.querySelectorAll('.js-file-upload__error-wrapper')
+  );
+
+  await t
+    .expect(await $errorWrapper.innerText)
+    .eql('You exceeded the maximum number of files');
+});
 
 fixture('File upload - maxSizeOfAllFilesKB prop').page(
   `${host}/iframe.html?id=molecules-file-upload--file-upload&knob-inputFileText=Upload%20file&knob-maxSizeOfSingleFileKB=100&knob-maxSizeOfAllFilesKB=1&knob-maxNumberOfFiles=10&knob-deleteStatusText=Delete&knob-addStatusText=Add%20more&knob-fileTooBigStatusText=File%20size%20exceeds%20maximum%20size&knob-filesTooBigStatusText=File%20sizes%20exceed%20maximum%20size&knob-tooManyFilesStatusText=You%20exceeded%20the%20maximum%20number%20of%20files&knob-orText=or&knob-infoText=Drag%20and%20drop%20to%20upload%20your%20file&knob-icon=cloud-upload&knob-headerText=The%20following%20files%20are%20being%20transferred:`

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -42,6 +42,7 @@ test('should upload file and switch view', async t => {
       .querySelector('axa-file-upload')
       .shadowRoot.querySelector('.js-file-upload__dropzone-file-overview')
   );
+  await $fileOverviewElem(); // Travis failed sometimes a line below. (Bug #1335)
   await t.expect($fileOverviewElem.exists).ok();
 });
 
@@ -185,7 +186,7 @@ test('should exceed maximum number of files', async t => {
     () => document.querySelector('axa-file-upload').shadowRoot
   ).find('.js-file-upload__img-figure');
 
-  await $figureElems();
+  await $figureElems(); // Travis failed sometimes a line below. (Bug #1335)
   await t.expect(await $figureElems.count).eql(1);
 
   const $errorWrapper = await Selector(() =>

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -151,9 +151,11 @@ test('should exceed maximum size of single file', async t => {
 });
 
 // eslint-disable-next-line jest/no-focused-tests
-fixture.only('File upload - maxNumberOfFiles prop').page(
-  `${host}/iframe.html?id=molecules-file-upload--file-upload&knob-inputFileText=Upload%20file&knob-maxSizeOfSingleFileKB=100&knob-maxSizeOfAllFilesKB=500&knob-maxNumberOfFiles=1&knob-deleteStatusText=Delete&knob-addStatusText=Add%20more&knob-fileTooBigStatusText=File%20size%20exceeds%20maximum%20size&knob-filesTooBigStatusText=File%20sizes%20exceed%20maximum%20size&knob-tooManyFilesStatusText=You%20exceeded%20the%20maximum%20number%20of%20files&knob-orText=or&knob-infoText=Drag%20and%20drop%20to%20upload%20your%20file&knob-icon=cloud-upload&knob-headerText=The%20following%20files%20are%20being%20transferred:`
-);
+fixture
+  .only('File upload - maxNumberOfFiles prop')
+  .page(
+    `${host}/iframe.html?id=molecules-file-upload--file-upload&knob-inputFileText=Upload%20file&knob-maxSizeOfSingleFileKB=100&knob-maxSizeOfAllFilesKB=500&knob-maxNumberOfFiles=1&knob-deleteStatusText=Delete&knob-addStatusText=Add%20more&knob-fileTooBigStatusText=File%20size%20exceeds%20maximum%20size&knob-filesTooBigStatusText=File%20sizes%20exceed%20maximum%20size&knob-tooManyFilesStatusText=You%20exceeded%20the%20maximum%20number%20of%20files&knob-orText=or&knob-infoText=Drag%20and%20drop%20to%20upload%20your%20file&knob-icon=cloud-upload&knob-headerText=The%20following%20files%20are%20being%20transferred:`
+  );
 test('should remove addMoreInputFile', async t => {
   const $inputFileInputElem = await Selector(() =>
     document

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -101,6 +101,7 @@ test('should delete image', async t => {
       .shadowRoot.querySelectorAll('.js-file-upload__img-figure')
   );
 
+  await $figureElemsBefore(); // Travis failed sometimes a line below. (Bug #1335)
   await t.expect(await $figureElemsBefore.count).eql(2); // one file + addMoreInputFile
 
   const $figureElem = await Selector(() =>
@@ -117,6 +118,7 @@ test('should delete image', async t => {
       .shadowRoot.querySelectorAll('.js-file-upload__img-figure')
   );
 
+  await $figureElemsAfter(); // Travis failed sometimes a line below. (Bug #1335)
   await t.expect(await $figureElemsAfter.count).eql(0); // zero because start view has no figures
 });
 
@@ -140,7 +142,9 @@ test('should exceed maximum size of single file', async t => {
       .shadowRoot.querySelectorAll('.js-file-upload__img-figure')
   );
 
+  await $figureElems(); // Travis failed sometimes a line below. (Bug #1335)
   await t.expect(await $figureElems.count).eql(2); // one file + addMoreInputFile
+
   const $fileName = await Selector(() =>
     document
       .querySelector('axa-file-upload')

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -177,7 +177,7 @@ test('should exceed maximum number of files', async t => {
 
   await t.expect($inputFileInputElem.exists).ok();
 
-  await t.setFilesToUpload($inputFileInputElem, [validFiles[0], validFiles[1]]);
+  await t.setFilesToUpload($inputFileInputElem, validFiles);
 
   const $figureElems = await Selector(
     () => document.querySelector('axa-file-upload').shadowRoot

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -170,6 +170,7 @@ test('should remove addMoreInputFile', async t => {
   await t.expect($addMoreInputFieldElem.exists).notOk();
 });
 
+// eslint-disable-next-line jest/no-focused-tests
 test.only('should exceed maximum number of files', async t => {
   const $inputFileInputElem = await Selector(
     () => document.querySelector(FILE_UPLOAD_TAG).shadowRoot,
@@ -178,6 +179,7 @@ test.only('should exceed maximum number of files', async t => {
 
   await t.expect($inputFileInputElem.exists).ok();
 
+  // eslint-disable-next-line no-console
   console.log(await $inputFileInputElem());
   await t.setFilesToUpload($inputFileInputElem, validFiles);
 
@@ -185,7 +187,9 @@ test.only('should exceed maximum number of files', async t => {
     () => document.querySelector('axa-file-upload').shadowRoot
   ).find('.js-file-upload__img-figure');
 
+  // eslint-disable-next-line no-console
   console.log('figure element');
+  // eslint-disable-next-line no-console
   console.log(await $figureElems());
 
   await t.expect(await $figureElems.count).eql(1);

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -32,6 +32,7 @@ test('should upload file and switch view', async t => {
         .querySelector('axa-file-upload')
         .shadowRoot.querySelector('axa-input-file .a-input-file__input') // TODO change to .js-
   );
+
   await t.expect($inputFileInputElem.exists).ok();
 
   await t.setFilesToUpload($inputFileInputElem, validFiles);
@@ -169,7 +170,7 @@ test('should remove addMoreInputFile', async t => {
   await t.expect($addMoreInputFieldElem.exists).notOk();
 });
 
-test('should exceed maximum number of files', async t => {
+test.only('should exceed maximum number of files', async t => {
   const $inputFileInputElem = await Selector(
     () => document.querySelector(FILE_UPLOAD_TAG).shadowRoot,
     { dependencies: { FILE_UPLOAD_TAG } }
@@ -177,11 +178,15 @@ test('should exceed maximum number of files', async t => {
 
   await t.expect($inputFileInputElem.exists).ok();
 
+  console.log(await $inputFileInputElem());
   await t.setFilesToUpload($inputFileInputElem, validFiles);
 
   const $figureElems = await Selector(
     () => document.querySelector('axa-file-upload').shadowRoot
   ).find('.js-file-upload__img-figure');
+
+  console.log('figure element');
+  console.log(await $figureElems());
 
   await t.expect(await $figureElems.count).eql(1);
 

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -177,8 +177,7 @@ test('should exceed maximum number of files', async t => {
 
   await t.expect($inputFileInputElem.exists).ok();
 
-  await t.setFilesToUpload($inputFileInputElem, validFiles[0]);
-  await t.setFilesToUpload($inputFileInputElem, validFiles[1]);
+  await t.setFilesToUpload($inputFileInputElem, [validFiles[0], validFiles[1]]);
 
   const $figureElems = await Selector(
     () => document.querySelector('axa-file-upload').shadowRoot

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -32,7 +32,6 @@ test('should upload file and switch view', async t => {
         .querySelector('axa-file-upload')
         .shadowRoot.querySelector('axa-input-file .a-input-file__input') // TODO change to .js-
   );
-
   await t.expect($inputFileInputElem.exists).ok();
 
   await t.setFilesToUpload($inputFileInputElem, validFiles);
@@ -150,12 +149,9 @@ test('should exceed maximum size of single file', async t => {
   await t.expect($fileName.innerText).eql('File size exceeds maximum size');
 });
 
-// eslint-disable-next-line jest/no-focused-tests
-fixture
-  .only('File upload - maxNumberOfFiles prop')
-  .page(
-    `${host}/iframe.html?id=molecules-file-upload--file-upload&knob-inputFileText=Upload%20file&knob-maxSizeOfSingleFileKB=100&knob-maxSizeOfAllFilesKB=500&knob-maxNumberOfFiles=1&knob-deleteStatusText=Delete&knob-addStatusText=Add%20more&knob-fileTooBigStatusText=File%20size%20exceeds%20maximum%20size&knob-filesTooBigStatusText=File%20sizes%20exceed%20maximum%20size&knob-tooManyFilesStatusText=You%20exceeded%20the%20maximum%20number%20of%20files&knob-orText=or&knob-infoText=Drag%20and%20drop%20to%20upload%20your%20file&knob-icon=cloud-upload&knob-headerText=The%20following%20files%20are%20being%20transferred:`
-  );
+fixture('File upload - maxNumberOfFiles prop').page(
+  `${host}/iframe.html?id=molecules-file-upload--file-upload&knob-inputFileText=Upload%20file&knob-maxSizeOfSingleFileKB=100&knob-maxSizeOfAllFilesKB=500&knob-maxNumberOfFiles=1&knob-deleteStatusText=Delete&knob-addStatusText=Add%20more&knob-fileTooBigStatusText=File%20size%20exceeds%20maximum%20size&knob-filesTooBigStatusText=File%20sizes%20exceed%20maximum%20size&knob-tooManyFilesStatusText=You%20exceeded%20the%20maximum%20number%20of%20files&knob-orText=or&knob-infoText=Drag%20and%20drop%20to%20upload%20your%20file&knob-icon=cloud-upload&knob-headerText=The%20following%20files%20are%20being%20transferred:`
+);
 test('should remove addMoreInputFile', async t => {
   const $inputFileInputElem = await Selector(() =>
     document

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -185,6 +185,7 @@ test('should exceed maximum number of files', async t => {
     () => document.querySelector('axa-file-upload').shadowRoot
   ).find('.js-file-upload__img-figure');
 
+  t.wait(1);
   await t.expect(await $figureElems.count).eql(1);
 
   const $errorWrapper = await Selector(() =>

--- a/src/components/20-molecules/file-upload/ui.test.js
+++ b/src/components/20-molecules/file-upload/ui.test.js
@@ -179,18 +179,11 @@ test('should exceed maximum number of files', async t => {
 
   await t.expect($inputFileInputElem.exists).ok();
 
-  // eslint-disable-next-line no-console
-  console.log(await $inputFileInputElem());
   await t.setFilesToUpload($inputFileInputElem, validFiles);
 
   const $figureElems = await Selector(
     () => document.querySelector('axa-file-upload').shadowRoot
   ).find('.js-file-upload__img-figure');
-
-  // eslint-disable-next-line no-console
-  console.log('figure element');
-  // eslint-disable-next-line no-console
-  console.log(await $figureElems());
 
   await t.expect(await $figureElems.count).eql(1);
 

